### PR TITLE
Refactor: Further schema improvements

### PR DESCRIPTION
**Pull request by timonback**
Split API and angular view models
Introduce AsyncApiMapperService to map the models
Implement stronger typescript typing

### CODE REVIEW

File: src/app/shared/asyncapi-mapper.service.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
+import { AsyncApi } from './models/asyncapi.model';
```
Comment (stavshamir) [inline src/app/shared/asyncapi-mapper.service.ts:1]: Good idea to extract this, maybe I will finally write some unit tests

File: src/app/shared/asyncapi.service.ts [hunk 2]
```diff
@@ -50,70 +20,7 @@ export class AsyncApiService {
         }
 
         return this.http.get<ServerAsyncApi>(Endpoints.docs).pipe(map(item => {
-            this.docs = this.toAsyncApi(item);
```
Comment (stavshamir) [inline src/app/shared/asyncapi.service.ts:53]: This line is important - without it each call to `getAsyncApis` (which should be renamed to `getAsyncApi`, I missed that in my previous PR) will trigger an actual http request.

File: src/app/shared/models/channel.model.ts [hunk 3]
```diff
@@ -1,20 +1,29 @@
+export const CHANNEL_ANCHOR_PREFIX = "#channel-"
 export interface Channel {
     name: string;
+    anchorIdentifier: string;
     description?: string;
     operation: Operation;
 }
 
+export type OperationType = "publish" | "subscribe";
```
Comment (stavshamir) [inline src/app/shared/models/channel.model.ts:9]: typescript is great :)

**Review by stavshamir — CHANGES_REQUESTED** [review]
Great improvements, see inline comments. Also, I tried playing with the preview and everything looks fine, except for the links to the components. The current deployed demo works, so it probably comes from this PR, I couldn't find out why though.

File: src/app/shared/asyncapi.service.ts (hunk 2 above)
Comment (timonback) [reply src/app/shared/asyncapi.service.ts:53]: Thanks for pointing this out. Now I start to understand the pattern here

**Comment by stavshamir** [comment]
Looks great, thanks

**Review by jrlambs — COMMENTED** [review]
this cleaned up a lot! love it!

**Pull request merged**